che-mount/entrypoint.sh

```diff
@@ -93,45 +107,95 @@ trap 'stop_sync' SIGHUP SIGTERM SIGINT
-status=$?
-if [ $status -ne 0 ]; then
-    error "ERROR: Fatal error occurred ($status)"
-    exit 1
-fi
-info "INFO: (che mount): Background sync continues every ${UNISON_REPEAT_DELAY_IN_SEC} seconds."
-info "INFO: (che mount): This terminal will block while the synchronization continues."
-info "INFO: (che mount): To stop, issue a SIGTERM or SIGINT, usually CTRL-C."
-
-# run application
-unison /mntssh /mnthost -batch -retry 10 -fat -silent -copyonconflict -auto -prefer=newer -repeat=${UNISON_REPEAT_DELAY_IN_SEC} -log=false > /dev/null 2>&1
+chmod 700 $HOME/.ssh/id_rsa
+if [ "${CHE_SYNC_AGENT}" = "true" ]; then
+    info "(che mount): Syncing ${SSH_USER}@${SSH_IP}:${SSH_PORT}/projects with workspace sync agent."
+	info "(che mount): Initial sync... Please wait."
+    START_TIME=$(date +%s)
+    unison /mnthost ssh://${SSH_USER}@${SSH_IP}:${SSH_PORT}//../../projects/ -batch -retry 10 \
+            -fat -silent -copyonconflict -auto -prefer=newer -log=false \
+            -sshargs '-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no'> /dev/null 2>&1 
+    status=$?
+	if [ $status -ne 0 ]; then
+	    error "ERROR: Fatal error occurred ($status)"
+	    exit 1
+	fi
+    ELAPSED_TIME=$(expr $(date +%s) - $START_TIME)
+    info "(che mount): Initial Unison sync took using sync agent $ELAPSED_TIME seconds."
+    info "(che mount): Background sync continues every ${UNISON_REPEAT_DELAY_IN_SEC} seconds."
+    info "(che mount): This terminal will block while the synchronization continues."
+    info "(che mount): To stop, issue a SIGTERM or SIGINT, usually CTRL-C."
+    while [ 1 ]
+    do
+        sleep ${UNISON_REPEAT_DELAY_IN_SEC}
+        unison /mnthost ssh://${SSH_USER}@${SSH_IP}:${SSH_PORT}//../../projects/ -batch -retry 10 \
+            -fat -silent -copyonconflict -auto -prefer=newer -log=false \
+            -sshargs '-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no'> /dev/null 2>&1   
+    done
+    status=$?
+	if [ $status -ne 0 ]; then
+	    error "ERROR: Fatal error occurred ($status)"
+	    exit 1
+	fi
+else
+    info "(che mount): Sync agent not detected using SSHFS."
+    info "(che mount): Mounting ${SSH_USER}@${SSH_IP}:/projects (${SSH_PORT}) with SSHFS."
+    sshfs ${SSH_USER}@${SSH_IP}:/projects /mntssh -p ${SSH_PORT} -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no
+    status=$?
+    if [ $status -ne 0 ]; then
+        error "Fatal error occurred ($status)"
+        exit 1
+    fi
+    info "(che mount): Successfully mounted ${SSH_USER}@${SSH_IP}:/projects (${SSH_PORT})"
+    info "(che mount): Initial sync...Please wait."
+    START_TIME=$(date +%s)
+    unison /mntssh /mnthost -batch -fat -silent -auto -prefer=newer -log=false > /dev/null 2>&1
+    status=$?
+	if [ $status -ne 0 ]; then
+	    error "ERROR: Fatal error occurred ($status)"
+	    exit 1
+	fi
+    ELAPSED_TIME=$(expr $(date +%s) - $START_TIME)
+    info "INFO: (che mount): Initial Unison sync took $ELAPSED_TIME seconds."
```

Comment:
Not sure if I understand your question. I think having initial sync time is helpful information for the user.